infrastructure/kafka/kafka.go

```diff
@@ -1,28 +1,67 @@
 package kafka
 
 import (
+	"context"
+	"errors"
+	"fmt"
+
 	kafka "github.com/segmentio/kafka-go"
 	"github.com/spf13/viper"
 )
 
-func NewResultsConsumer(cfg *viper.Viper) *kafka.Reader {
+func NewConsumer(cfg *viper.Viper, topic string) (*kafka.Reader, error) {
 	consumer := kafka.NewReader(kafka.ReaderConfig{
 		Brokers:     cfg.GetStringSlice("Kafka_Brokers"),
-		Topic:       cfg.GetString("Kafka_Results_Topic"),
+		Topic:       topic,
 		GroupID:     cfg.GetString("Kafka_Group_ID"),
 		StartOffset: cfg.GetInt64("Kafka_Consumer_Offset"),
 	})
 
-	return consumer
+	err := verifyTopic(cfg.GetStringSlice("Kafka_Brokers"), topic)
+
+	return consumer, err
 }
 
-func NewConnectionsConsumer(cfg *viper.Viper) *kafka.Reader {
-	consumer := kafka.NewReader(kafka.ReaderConfig{
-		Brokers:     cfg.GetStringSlice("Kafka_Brokers"),
-		Topic:       cfg.GetString("Kafka_Connections_Topic"),
-		GroupID:     cfg.GetString("Kafka_Group_ID"),
-		StartOffset: cfg.GetInt64("Kafka_Consumer_Offset"),
-	})
+func NewConsumerEventLoop(
+	ctx context.Context,
+	consumer *kafka.Reader,
+	handler func(context.Context, kafka.Message),
+	errors chan<- error,
+) (start func()) {
+	return func() {
+		for {
+			m, err := consumer.ReadMessage(ctx)
+			if err != nil {
+				fmt.Println(err)
+				errors <- err
+			}
+			handler(ctx, m)
+		}
+	}
+}
+
+func verifyTopic(brokers []string, topic string) error {
```

Comment:
Interesting - I am curious how you got to need this verification. Does `kafka.NewReader()` not complain if you point it to a non-existent topic?

Reply:
Well I thought this would be a bit more useful until I realized that kafka was automatically creating the new topic if it didn't exist already. I believe this is just a configuration option and I _assume_ this isn't the case in our deployed kafka. Looking into the code it does seem like it'll complain if the topic doesn't exist so I'm going to remove this verification for now.

Reply:
> Well I thought this would be a bit more useful until I realized that kafka was automatically creating the new topic if it didn't exist already. I believe this is just a configuration option and I assume this isn't the case in our deployed kafka.
That's correct

> Looking into the code it does seem like it'll complain if the topic doesn't exist so I'm going to remove this verification for now.
Yeah I wasn't sure but IIRC it does complain.